README.md

```diff
@@ -1,87 +1,121 @@
+```ERB
+<%= form_for(@post) do |f| %>
+```
 
-second, with that, params[:post] returns a nice hash {:title => "Post Title"}. Why is that useful? Becasue we can mass-assign the updated post attributes to update with @post.update(params[:post]) (we have to account for strong parameters, either telling them to shut it off, providing code, or teaching them about it here).
+The `|f|` is an iterator variable that we can use on the new form object that will allow us to dynamically assign form field elements to each of the `post` data attributes, along with auto filling the values for each field. We get this `ActionView` functionality because we're using the `form_form` method and that gives us access to the `FormBuilder` module in Rails ([Documentation](http://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html)). Inside of the form we can now refactor the fields:
 
-That's pretty nice.
+```ERB
+<label>Post title:</label><br>
+<%= f.text_field :title %><br>
 
-Let's go back to our form and add content, the exact situation we said made our code disproportionate earlier. before, adding a new field required two update, one to the form and one to the controller action. watch what happens now.
+<label>Post Description</label><br>
+<%= f.text_area :description %><br>
+```
 
-form_for(@post) do |form_builder| <%= form_builder.text_field :title %> <%= form_builder.text_area :content %> <%= form_builder.submit %> end
+Isn't that much cleaner? Notice how we no longer have to pass in the values manually? By passing in the attribute as a symbol (e.g. `:title`) that will automatically tell the form field what model attribute to bind to, it also is what auto fills the values for us. Now let's refactor the submit button, instead of `<%= submit_tag "Submit Post" %>` we can change it to:
 
-submit the form and we don't need to update the controller action because of mass-assignment.
+```ERB
+<%= f.submit %>
+```
 
-That in a nutshell is how form_for works.
+Lastly, `form_for` also automatically sets the `authenticity_token` value for us, so we can remove the `<%= hidden_field_tag :authenticity_token, form_authenticity_token %>` line completely. Our new form will look something like this:
 
-other form builders, select_tag, hidden_tag, etc.
-options to pass to form_for, most important is URL
+```ERB
+<h3>Post Form</h3>
+
+<%= form_for(@post) do |f| %>
+  <label>Post title:</label><br>
+  <%= f.text_field :title %><br>
+
+  <label>Post Description</label><br>
+  <%= f.text_area :description %><br>
+  
+  <%= f.submit %>
+<% end %>
+```
+
+Our refactor work isn't quite done, if you had previously created a `PUT` route like we did in the `form_tag` lesson, we'll need to change that to a `PATCH` method since that is the HTTP verb that `form_for` utilizes, we can make that change in the `config/routes.rb` file:
```

Comment:
Can you explain the diff between PUT and PATCH?  Pretty much PUT expects a full formed object put, but PATCH just expects the changed field (right?)